packages/preview/src/components/@core/sidebar/index.tsx

```diff
@@ -30,6 +33,11 @@ export default function Sidebar() {
     });
   };
 
+  const debouncedSearchHandler = useMemo(
+    () => debounce(onSearch, DEBOUNCE_WAIT_TIME),
+    []
+  );
```

Comment:
You don't need the useMemo part. It is wrong in this case since onSearch is re-created on each render.

This is enough 
```tsx
const debouncedSearchHandler = debounce(onSearch, DEBOUNCE_WAIT_TIME);
```